react/game/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,112 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.scss';
 import Board from './components/Board';
 
-function Game() {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [
+      {
+        squares: Array(9).fill(null)
+      }
+    ],
+    xIsNext: true,
+    stepNumber: 0,
+    winner: null
+  };
+
+  handleClick = i => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (this.state.winner || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          squares
+        }
+      ]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext,
+      winner: this.calculateWinner(squares)
+    });
+  };
+
+  jumpTo = step => {
+    this.setState({
```

Comment:
U must use ` prevState`. Check this https://teamtreehouse.com/community/react-docs-now-recommends-using-function-with-prevstate-inside-of-setstate

Reply:
I will have it in my mind from onwards.